main/broadcasters.c

```diff
@@ -763,12 +763,90 @@ static broadcaster_ops_t atc1441_temp_hum_ops = {
     .metadata_get = atc1441_temp_hum_metadata_get,
 };
 
+/* PVVX Firmware for the Xiaomi Thermometer LYWSD03MMC Temperature and
+ * Humidity Sensor, see: https://github.com/pvvx/ATC_MiThermometer */
+
+#define PVVX_TEMP_HUM_SERVICE_UUID 0x181A
+
+typedef struct {
+//    uint16_t not_used;
+    uint16_t service_uuid;
+    mac_addr_t mac;
+    int16_t temp;
+    uint16_t humid;
+    uint16_t battery_mv;
+    uint8_t battery_percent;
+    uint8_t message_counter;
+    uint8_t flags;
+} __attribute__((packed)) pvvx_temp_hum_t;
+
+static pvvx_temp_hum_t *pvvx_temp_hum_data_get(uint8_t *adv_data,
+    uint8_t adv_data_len, uint8_t *pvvx_temp_hum_len)
+{
+    uint8_t len;
+    uint8_t *data = esp_ble_resolve_adv_data(adv_data,
+        ESP_BLE_AD_TYPE_SERVICE_DATA, &len);
+
+    if (pvvx_temp_hum_len)
+        *pvvx_temp_hum_len = len;
+
+    return (pvvx_temp_hum_t *)data;
+}
+static int pvvx_temp_hum_is_broadcaster(uint8_t *adv_data,
+    size_t adv_data_len)
+{
+    uint8_t len;
+    pvvx_temp_hum_t *pvvx_data = pvvx_temp_hum_data_get(adv_data,
+        adv_data_len, &len);
+
+    if (len < sizeof(pvvx_temp_hum_t) ||
```

Comment:
`<` or `!=`?
Are you expecting variable length messages?

Reply:
Not expecting variable length, but was following convention of ATC and other broadcasters. Happy to change if you'd prefer.